industrial_ci/src/tests/debians.sh

```diff
@@ -0,0 +1,64 @@
+#!/bin/bash
+
+# Copyright (c) 2021, Mathias Lüdtke
+# All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Based on https://github.com/ros-industrial/industrial_ci/issues/697#issuecomment-876293987
+
+function prepare_debians() {
+  true
+}
+
+function build_and_install_debian() (
+    set -e
+    local pkg_path=$1
+    shift
+    cd "$pkg_path"
+    ici_quiet_echo bloom-generate rosdebian --ros-distro="$ROS_DISTRO" --debian-inc="ici~"
+    ici_quiet_echo dpkg-buildpackage -b -uc -us
+    mv ../*.{buildinfo,changes,deb} .
+    ici_quiet_echo ici_asroot dpkg -i ./*.deb
+)
+
+function run_debians() {
+    export BUILDER=colcon
+    ici_source_builder
+    ici_run "${BUILDER}_setup" ici_quiet builder_setup
+    ici_run "setup_bloom" ici_quiet ici_install_pkgs_for_command bloom-generate python3-bloom debhelper
+    ici_run "setup_dpkg_dev" ici_quiet ici_install_pkgs_for_command dpkg-buildpackage dpkg-dev
+
+    ici_run "setup_rosdep" ici_setup_rosdep
+
+    if [ ! -f "$_ROS_KEYRING" ]; then
+      ici_run "setup_gpg_key" ici_setup_gpg_key
+    fi
+
+    local sourcespace
+    ici_make_temp_dir sourcespace
+
+    for name in upstream target downstream; do
+      local sources=()
+      local current="$sourcespace/$name"
+      ici_parse_env_array sources "${name^^}_WORKSPACE"
+      if [ -n "${sources[*]}" ]; then
+        ici_run "prepare_${name}_sourcespace" ici_prepare_sourcespace "$current" "${sources[@]}"
+        ici_run "install_${name}_dependencies" ici_quiet ici_install_dependencies "$UNDERLAY" "$ROSDEP_SKIP_KEYS" "$sourcespace"
+
+        while read -r -a pkg; do
+          ici_run "${pkg[0]}_debian" build_and_install_debian "$current/${pkg[1]}"
```

Comment:
this would still all be run in the same root filesystem, right?
The point of sbuild is to have a minimal chroot (container) (only essential packages installed) to find missing dependencies. With your setup if you have a repo with multiple ROS packages and package A ab B have a dependency X but only A declares it, packages B would still build if the dependency was installed for A before. That would not happen with sbuild nor work on the ROS build farm (and was the point in #697).

Reply:
I see. Not sure how to solve this. `sbuild` is just too complicated to set-up (incl. some manual steps).
The ROS pre-release test builds each package in a clean environment already, so I wonder why it did not fail in the first place.

Reply:
which pre-release tests do you mean?
http://prerelease.ros.org/ does not, afaik.

Reply:
Regarding sbuild being too complicated, you could spin up a new docker container for every package

Reply:
> which pre-release tests do you mean?

http://prerelease.ros.org/ does not, afaik.

This one, but with without the need to register it in rosdistro. 
I thought that it creates a new container (actually, new containers) for each package, but after digging into the log files this might not be the case :/

> Regarding sbuild being too complicated, you could spin up a new docker container for every package

Yes, I will try this next. Together with a shared, local repository.

Reply:
The latest version does this, at least locally.
I will provide a test case as well later.